.github/workflows/tests.yml

```diff
@@ -0,0 +1,277 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+      - release
+  pull_request:
+    branches:
+      - '*'
+  schedule:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule
+    # > Scheduled workflows run on the latest commit on the default or base branch
+    # i.e. this can only run on master
+    - cron:  '0 11 * * *'
+
+# The set of available runners on Github's infra is generous but
+# will never be complete: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
+# to cover other other platforms Linux distros, we use docker or WSL images.
+# These are therefore not perfectly accurate tests, but hopefully good enough.
+
+# We haven't figured out how to factor this -- to avoid the boilerplate. It seems impossible.
+# Github Actions is just not flexible enough. The best it has is the build matrix, but:
+# - the bad null case: if the build matrix generates no jobs, the UI prints a placeholder with the unevaluated `name:` expression like `${{matrix.name || matrix.os || 'Nightlies' }}`
+# - you cannot refer to the matrix parameters in a job's `if:`, so it is impossible to decide whether to run a job or not; it is possible to do so in a *step*'s `if:`. See https://github.community/t/conditional-matrices/17206/2.
+#   ideally we would have a build matrix with all the platforms covered
+#   ```
+#   if: ${{ !matrix.nightly || (github.event_name == 'schedule' || github.event_name == 'push') }}
+#   ```
+#   but it's not allowed; that second half is legal but `matrix.nightly` is not.
+# - there's no way to mix the docker-based platforms with the windows ones in the same build matrix anyway, because specifying `container:`, even with an empty/undefined string, tries blindly to use the docker scripts
```

Comment:
Now that I see it, `(github.event_name == 'schedule' || github.event_name == 'push')` seems very clear and descriptive, and not that different from what we were doing in Travis. I actually like doing this instead of `matrix.nightly`.

So, I'm not sure we need to anticipate `matrix.nightly` anymore, and I think we could remove the comments/TODOs.